deps/rmutil/periodic.c

```diff
@@ -107,10 +107,16 @@ void RMUtilTimer_ForceInvoke(struct RMUtilTimer *t) {
 }
 
 int RMUtilTimer_Signal(struct RMUtilTimer *t) {
-  return pthread_cond_signal(&t->cond);
+  pthread_mutex_lock(&t->lock);
+  int ret = pthread_cond_signal(&t->cond);
+  pthread_mutex_unlock(&t->lock);
+  return ret;
 }
 
 int RMUtilTimer_Terminate(struct RMUtilTimer *t) {
+  pthread_mutex_lock(&t->lock);
```

Comment:
I guess that here the terminate is rare called so the locking will not effect performance?

Reply:
only on shutdown event with RS_GLOBAL_DTORS=1